### ISSUE TITLE
Just updated to 1.6.2 - $digest already in progress error.

### ISSUE THREAD

**Issue by Jerczu** [angular-ui/ui-scroll#168]
```
angular.js:14525 Error: [$rootScope:inprog] $digest already in progress
http://errors.angularjs.org/1.6.4/$rootScope/inprog?p0=%24digest
    at angular.js:66
    at beginPhase (angular.js:18524)
    at Scope.$digest (angular.js:17947)
    at Scope.$apply (angular.js:18269)
    at adjustBufferAfterFetch (ui-scroll.js:359)
    at ui-scroll.js:403
    at processQueue (angular.js:16832)
    at angular.js:16876
    at Scope.$digest (angular.js:17971)
    at Scope.$apply (angular.js:18269)
```

Not really sure what is the issue here.

**Comment by MisterDev** [comment]
Same here, in my case it is triggered by `self.adapter.applyUpdates`

**Comment by Jerczu** [comment]
I use dataSource on scope pretty much the same way as one in examples.

**Comment by Jerczu** [comment]
Ok I fixed it locally
`
function adjustBufferAfterFetch(rid) {
  | var updates = updateDOM();
  |  
  | // We need the item bindings to be processed before we can do adjustment
  | $scope.$apply(function () {
  | return $timeout(function () {
  | // show elements after data binging has been done
  | updates.inserted.forEach(function (w) {
  | return w.element.removeClass('ng-hide');
  | });
  | updates.prepended.forEach(function (w) {
  | return w.element.removeClass('ng-hide');
  | });
  |  
  | viewport.adjustScrollTopAfterPrepend(updates);
  |  
  | if (isInvalid(rid)) {
  | return;
  | }
  |  
  | updatePaddings(rid, updates);
  | enqueueFetch(rid, updates);
  | pending.shift();
  |  
  | if (pending.length) fetch(rid);else {
  | adapter.loading(false);
  | bindEvents();
  | adapter.calculateProperties();
  | }
  | });
  | });
  | }
`
instead of using $apply I changed it to $applyAsync and it seem to work just fine now.

**Comment by Jerczu** [comment]
I've made a PR that looks to fix the issue. I left the $timeouts in (though $applyAsync doesn't require them Travis doesn't seem to like it without it)

**Comment by dhilt** [comment]
@Jerczu @MisterDev I accepted PR, thanks! But also I've just merged 'sync' branch into 'master' and 1.7.0-rc.1 is available via `npm install angular-ui-scroll`. 1.7.0-rc.1 fixes that issue in the same manner that your solution. Also the build process is fixed by 1.7.0.-rc.1. So please update the lib and try it.
